tests/qnn/test_keras.py

```diff
@@ -335,16 +470,49 @@ def test_gradients(self, get_circuit, output_dim, n_qubits):
 
         g_layer = tape.gradient(out_layer, layer.trainable_variables)
 
+        circuit_weights = layer.trainable_variables.copy()
+        circuit_weights[3] = 1.0 * circuit_weights[3]
+        circuit_weights[5] = 1.0 * circuit_weights[5]
+
         with tf.GradientTape() as tape:
-            out_circuit = c(x[0], *layer.trainable_variables)
+            out_circuit = c(x[0], *circuit_weights)
 
         g_circuit = tape.gradient(out_circuit, layer.trainable_variables)
 
         for i in range(len(out_layer)):
             assert np.allclose(g_layer[i], g_circuit[i])
 
+    @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
+    def test_backprop_gradients(self, mocker):
+        """Test if KerasLayer is compatible with the backprop diff method."""
+        if not qml.tape_mode_active():
+            pytest.skip("This functionality is only supported in tape mode.")
+
+        dev = qml.device("default.qubit.tf", wires=2)
+
+        @qml.qnode(dev, interface="tf", diff_method="backprop")
+        def f(inputs, weights):
+            qml.templates.AngleEmbedding(inputs, wires=range(2))
+            qml.templates.StronglyEntanglingLayers(weights, wires=range(2))
+            return [qml.expval(qml.PauliZ(i)) for i in range(2)]
+
+        weight_shapes = {"weights": (3, 2, 3)}
+
+        qlayer = qml.qnn.KerasLayer(f, weight_shapes, output_dim=2)
+
+        inputs = tf.ones((4, 2))
+
+        with tf.GradientTape() as tape:
+            out = tf.reduce_sum(qlayer(inputs))
+
+        spy = mocker.spy(qml.tape.tapes.QubitParamShiftTape, "jacobian")
+
+        grad = tape.gradient(out, qlayer.trainable_weights)
+        assert grad is not None
+        spy.assert_not_called()
+
 
-@pytest.mark.parametrize("interface", qml.qnodes.decorator.ALLOWED_INTERFACES)
+@pytest.mark.parametrize("interface", ["autograd", "torch", "tf"])
```

Comment:
How come this was updated? Was the old way not sufficient? Not suited to testing both old and new modes?

Reply:
Using `interface="numpy"` has been deprecated for a while; ever since we introduced TF/Torch support. The supported name is now `interface="autograd"`. In the new tape-mode, attempting to do `interface="numpy"` will now raise an error.

We could uses the new location of the allowed interface names:
```python
@pytest.mark.parametrize("interface", qml.tape.QNode.INTERFACE_MAP)
```

I always tend to prefer being explicit in the tests, though. Better that parametrized data is test-specific, rather than dependent on logic/implementation details from the codebase. If, at any point a new interface is added, ideally we should be notified to update tests due to reduced coverage.